### PR TITLE
Create Python script for indexing netcdf files

### PR DESCRIPTION
**Pull request by rod-glover**
Translate and update original indexing R script to Python.

### CODE REVIEW

**Comment by rod-glover** [comment]
This script depends on changes to `nchelpers` which at this moment have not yet been merged and published to our PyPI. 
All tests currently pass in development environment, which has modified nchelpers code installed locally.

**Comment by jameshiebert** [comment]
I'd prefer to get the compatible version ``nchelpers`` released, on our PyPI and required by this PR before merging. I'll [head over there](https://github.com/pacificclimate/nchelpers/pull/13) and take a look.

**Comment by rod-glover** [comment]
Most def. I should have been more explicit about that. And there is perhaps some discussion to be had about what belongs or doesn't in nchelpers. I stopped migrating helper functions from here to there in case we decided against certain ones.

File: mm_cataloguer/tests/test_index.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+# Helper functions for defining tests
+
+def conditional(f, false_value=None):
+    """Return a function that, dependent on an additional boolean keyword parameter `invoke`,
+     either invokes and returns the value of the argument function 
+     or else returns the value of the argument `false_value`.
+
+     Argument function is invoked with positional arguments passed to returned function.
+     """
+    def wrapper(*args, invoke=True):
```
Comment (jameshiebert) [inline mm_cataloguer/tests/test_index.py]: This syntax isn't valid in py27. We'll need to use:
```python
def wrapper(*args, **kwargs):
    invoke = kwargs.get('invoke', True)
```

**Comment by jameshiebert** [comment]
Looks like you still have a couple failing tests (to do with the level sets).

**Comment by rod-glover** [comment]
Forgot to add `modelmeta/data/tiny_hydromodel_gcm.nc` to `setup.py`.

**Comment by jameshiebert** [comment]
All right! I'm going to go ahead and merge this then.

**Pull request merged**